languages/python/sqlalchemy-oso/sqlalchemy_oso/roles2.py

```diff
@@ -83,6 +83,67 @@ def user_in_role_query(
     return query
 
 
+def user_in_role_query(
```

Comment:
So this gets both directly assigned and implied roles for the user?

Reply:
Yep, it does basically the same thing as is_allowed but starts from a specific role instead of from any role that contains a specific permission.